From 328b4700a4dc14e797d5d468ee1ef250e4200697 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Thu, 21 Mar 2019 10:19:40 +0100
Subject: [PATCH] python: use prefix from info

On exherbo we pass prefix=/usr because datadir cannot be in a parent directory.
This results in some stuff being installed in /usr/lib/python.
Use the prefix from python config instead of the one we pass to meson

Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
---
 mesonbuild/modules/python.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/modules/python.py b/mesonbuild/modules/python.py
index 049c457f..6847bb87 100644
--- a/mesonbuild/modules/python.py
+++ b/mesonbuild/modules/python.py
@@ -291,8 +291,8 @@ class PythonInstallation(ExternalProgramHolder):
         ExternalProgramHolder.__init__(self, python)
         self.interpreter = interpreter
         self.subproject = self.interpreter.subproject
-        prefix = self.interpreter.environment.coredata.get_builtin_option('prefix')
         self.variables = info['variables']
+        prefix = self.variables['prefix']
         self.paths = info['paths']
         install_paths = info['install_paths']
         self.platlib_install_path = os.path.join(prefix, install_paths['platlib'][1:])
-- 
2.21.0

